packages/container/lib/container.js

```diff
@@ -127,7 +141,73 @@ Container.prototype = {
    */
   lookupFactory(fullName, options) {
     assert('fullName must be a proper full name', this.registry.validateFullName(fullName));
-    return factoryFor(this, this.registry.normalize(fullName), options);
+
+    if (isFeatureEnabled('ember-no-double-extend')) {
```

Comment:
I think this check should be `ember-factory-for`?

Additionally it seems like you would pass the boolean as an argument to `deprecate`, doesn't really matter though :-p